### PR TITLE
fix(condo): DOMA-6083 added fields, fixed/refactored/added tests

### PR DESCRIPTION
**Pull request by akarjakin**
minapps on mobile side require resident id to work properly with sendAppPushMessage, so we decided that it should  receive residentId.

### CODE REVIEW

File: packages/codegen/generate.test.utils.js [hunk 1]
```diff
@@ -39,21 +58,21 @@ function _renderDeveloperFriendlyErrorMessage (data, errors, context) {
     }
 
     if (get(context, 'variables')) {
-        msg.push(`VARIABLES: ${JSON.stringify(context.variables)}`)
+        msg.push(`VARIABLES: ${JSON.stringify(context.variables, getCycleReplacer())}`)
```
Comment (pahaz) [inline packages/codegen/generate.test.utils.js:61]: it looks strange. You don't have a case for cycles inside context.variables

File: apps/condo/domains/miniapp/schema/SendAppPushMessageService.js [hunk 2]
```diff
@@ -87,16 +123,55 @@ const SendAppPushMessageService = new GQLCustomSchema('SendAppPushMessageService
...
+                    /** resident must belong to the user */
+                    const residentWhere = {
+                        id: residentId,
+                        user: { id: user.id },
+                        deletedAt: null,
+                    }
+                    const residentExisted = await Resident.getOne(context, residentWhere)
+
+                    if (!residentExisted) throw new GQLError(ERRORS.RESIDENT_NOT_FOUND, context)
+                } else {
```
Comment (pahaz) [inline apps/condo/domains/miniapp/schema/SendAppPushMessageService.js:156]: It looks like a security issue. Are you trying to fake an actual resident and bypass the access?

File: apps/condo/domains/common/utils/JSON.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+/**
+ * Replaces cyclic links within object for JSON.stringify to be able to serialize object properly
+ * usage: JSON.stringify(obj, getCycleReplacer(), 2)
+ * @returns {function(*, *=): *}
+ */
+const getCycleReplacer = () => {
```
Comment (pahaz) [inline apps/condo/domains/common/utils/JSON.js:6]: it's not a part of this PR. move to another

File: apps/condo/domains/miniapp/schema/SendAppPushMessageService.js [hunk 4]
```diff
@@ -71,7 +107,7 @@ const SendAppPushMessageService = new GQLCustomSchema('SendAppPushMessageService
         },
         {
             access: true,
-            type: 'input SendAppPushMessageData { body: String!, title: String, B2CAppContext: String }',
+            type: 'input SendAppPushMessageData { body: String!, title: String, B2CAppContext: String, resident: ResidentWhereUniqueInput, property: PropertyWhereUniqueInput, unitType: String, unitName: String }',
```
Comment (pahaz) [inline apps/condo/domains/miniapp/schema/SendAppPushMessageService.js:110]: It's strange you should add some `meta` field for any app-specific context and don't add resident, property or any other fields

**Comment by pahaz** [comment]
It looks questionable. I don't understand what are you trying to solve.

File: apps/condo/domains/miniapp/schema/SendAppPushMessageService.js [hunk 5]
```diff
@@ -2,20 +2,30 @@
  * Generated by `createservice miniapp.SendAppPushMessageService --type mutations`
  */
 
+const { get } = require('lodash')
+
 const { GQLError, GQLErrorCode: { BAD_USER_INPUT, FORBIDDEN } } = require('@open-condo/keystone/errors')
 const { checkDvAndSender } = require('@open-condo/keystone/plugins/dvAndSender')
 const { GQLCustomSchema } = require('@open-condo/keystone/schema')
 
+
```
Comment (Kekmus) [inline apps/condo/domains/miniapp/schema/SendAppPushMessageService.js:11]:
```suggestion
```

File: apps/condo/domains/miniapp/schema/SendAppPushMessageService.test.js [hunk 6]
```diff
@@ -27,235 +27,275 @@ const { getRandomTokenData, getRandomFakeSuccessToken } = require('@condo/domain
...
-    beforeEach( async () => {
+    beforeAll( async () => {
         admin = await makeLoggedInAdminClient()
+        residentClient = await makeClientWithResidentAccessAndProperty()
+
+        const [b2c] = await createTestB2CApp(admin)
+        const [residentData] = await createTestResident(admin, residentClient.user, residentClient.property)
+
+        user = residentClient.user
+        property = residentClient.property
```
Comment (Kekmus) [inline apps/condo/domains/miniapp/schema/SendAppPushMessageService.test.js:55]: Is property being used somewhere?

File: packages/codegen/generate.test.utils.js (hunk 1 above)
Comment (akarjakin) [reply packages/codegen/generate.test.utils.js:61]: errors have cyclick links in some cases

File: apps/condo/domains/miniapp/schema/SendAppPushMessageService.test.js [hunk 7]
```diff
@@ -27,235 +27,275 @@ const { getRandomTokenData, getRandomFakeSuccessToken } = require('@condo/domain
...
 
             expect(message.id).toMatch(UUID_RE)
         })
 
         it('Support can SendAppPushMessageService', async () => {
             const supportClient = await makeClientWithSupportUser()
-            const [user] = await createTestUser(admin)
             const [b2c] = await createTestB2CApp(admin)
+            await waitFor(async () => {
+                const [message] = await sendAppPushMessageByTestClient(supportClient, {
```
[the remark this answers is not in the document]
Comment (akarjakin) [reply apps/condo/domains/miniapp/schema/SendAppPushMessageService.test.js:77]: removed, it was for debugiing

File: apps/condo/domains/miniapp/schema/SendAppPushMessageService.js (hunk 2 above)
Comment (akarjakin) [reply apps/condo/domains/miniapp/schema/SendAppPushMessageService.js:156]: not at all. miniapp backend sometimes would have resident id and other times property id and unitType/unitName, so I check here that it resolved to the resident, connected to the user.

File: apps/condo/domains/common/utils/JSON.js (hunk 3 above)
Comment (akarjakin) [reply apps/condo/domains/common/utils/JSON.js:6]: ok

**Review by Kekmus — APPROVED** [review]

File: packages/codegen/generate.test.utils.js [hunk 8]
```diff
@@ -29,6 +29,25 @@ function _checkClient (client) {
...
+ * Replaces cyclic links within object for JSON.stringify to be able to serialize object properly
+ * usage: JSON.stringify(obj, getCycleReplacer(), 2)
+ * @returns {function(*, *=): *}
+ */
+const getCycleReplacer = () => {
+    const visited = new WeakSet()
+
+    return (key, value) => {
+        if (typeof value === 'object' && value !== null) {
+            if (visited.has(value)) return
```
Comment (pahaz) [inline packages/codegen/generate.test.utils.js:42]: '[CYCLE OBJECT]'

File: apps/condo/domains/miniapp/schema/SendAppPushMessageService.js (hunk 4 above)
Comment (akarjakin) [reply apps/condo/domains/miniapp/schema/SendAppPushMessageService.js]: fixed

File: packages/codegen/generate.test.utils.js (hunk 8 above)
Comment (akarjakin) [reply packages/codegen/generate.test.utils.js]: fixed

File: apps/condo/domains/miniapp/schema/MessageAppBlackList.test.js [hunk 9]
```diff
@@ -126,36 +125,4 @@ describe('MessageAppBlackList', () => {
             })
         })
     })
-
```
Comment (akarjakin) [inline apps/condo/domains/miniapp/schema/MessageAppBlackList.test.js:129]: moved to ./SendB2CAppPushMessageService.test.js

File: apps/condo/domains/miniapp/schema/SendB2CAppPushMessageService.js [hunk 10]
```diff
@@ -54,59 +72,80 @@ const ERRORS = {
...
-            schema: 'sendAppPushMessage(data: SendAppPushMessageInput!): SendAppPushMessageOutput',
-            resolver: async (parent, args, context, info, extra = {}) => {
+            access: access.canSendB2CAppPushMessage,
+            schema: 'sendB2CAppPushMessage(data: SendB2CAppPushMessageInput!): SendB2CAppPushMessageOutput',
+            resolver: async (parent, args, context) => {
                 const { data: argsData } = args
-                const { dv, sender, app, user, type, data, uniqKey } = argsData
+                const { dv, sender, app, user, resident, type, data, uniqKey } = argsData
                 const { B2CAppContext } = data
+                const residentId = get(resident, 'id')
```
Comment (sitozzz) [inline apps/condo/domains/miniapp/schema/SendB2CAppPushMessageService.js:122]: Whether resident object may not contain the given field? Why not just throw an error without additional db query right after this line?

File: apps/condo/domains/miniapp/schema/SendB2CAppPushMessageService.js [hunk 11]
```diff
@@ -1,21 +1,29 @@
...
 const { User } = require('@condo/domains/user/utils/serverSchema')
 const { RedisGuard } = require('@condo/domains/user/utils/serverSchema/guards')
 
-const { USER_NOT_FOUND_ERROR, APP_NOT_FOUND_ERROR, APP_BLACK_LIST_ERROR } = require('../constants')
+const {
+    USER_NOT_FOUND_ERROR, RESIDENT_NOT_FOUND_ERROR,
+    PROPERTY_NOT_FOUND_ERROR,
+    APP_NOT_FOUND_ERROR, APP_BLACK_LIST_ERROR,
+    RESIDENT_OR_PROPERTY_ID_MISSING_ERROR,
+} = require('../constants')
```
Comment (sitozzz) [inline apps/condo/domains/miniapp/schema/SendB2CAppPushMessageService.js:26]: Absolute import)

File: apps/condo/domains/miniapp/gql.js [hunk 12]
```diff
@@ -18,9 +18,9 @@ const ALL_MINI_APPS_QUERY = gql`
     }
 `
 
-const SEND_APP_PUSH_MESSAGE_MUTATION = gql`
-    mutation sendAppPushMessage ($data: SendAppPushMessageInput!) {
-        result: sendAppPushMessage(data: $data) { id status }
+const SEND_B2C_APP_PUSH_MESSAGE_MUTATION = gql`
+    mutation sendB2CAppPushMessage ($data: SendB2CAppPushMessageInput!) {
```
Comment (sitozzz) [inline apps/condo/domains/miniapp/gql.js:22]: Why is this renaming necessary? Could anyone use this mutation?

File: apps/condo/domains/miniapp/schema/SendB2CAppPushMessageService.js (hunk 10 above)
Comment (akarjakin) [reply apps/condo/domains/miniapp/schema/SendB2CAppPushMessageService.js:122]: that left of previous state, when resident was non required field. fixed.

File: apps/condo/domains/miniapp/gql.js (hunk 12 above)
Comment (akarjakin) [reply apps/condo/domains/miniapp/gql.js:22]: There will be other service for B2B messages when required. This renaming was discussed and approved with @pahaz  :)

**Review by sitozzz — APPROVED** [review]

**Review by pahaz — APPROVED** [review]

**Pull request merged**